src/EditorFeatures/Core/GoToDefinition/DefinitionItemService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Editor.GoToDefinition
+{
+    internal abstract class AbstractDefinitionItemService : ForegroundThreadAffinitizedObject, IDefinitionItemService, ILanguageService
+    {
+        public async Task<(string, ImmutableArray<DefinitionItem>)> GetDefinitionsAsync(Document document, int position, bool thirdPartyNavigationAllowed, CancellationToken cancellationToken)
+        {
+            var service = document.GetLanguageService<AbstractGoToDefinitionSymbolService>();
+            var symbol = await service.FindSymbolAsync(document, position, cancellationToken).ConfigureAwait(false);
+
+            if (symbol == null)
+            {
+                return default((string, ImmutableArray<DefinitionItem>));
+            }
+
+            var definitions = await Task.Factory.StartNew(() => GoToDefinitionHelpers.GetDefinitions(symbol, document.Project, cancellationToken, thirdPartyNavigationAllowed: thirdPartyNavigationAllowed), cancellationToken,
+                TaskCreationOptions.None,
+                ForegroundTaskScheduler).ConfigureAwait(false);
+            return (FindUsagesHelpers.GetDisplayName(symbol), definitions);
```

Comment:
Hrmm.. DefinitionItems already have a "NameDisplayParts".  Not sure why we need this.